### DISCUSSION TITLE
Lox member call to access class member failed

### DISCUSSION THREAD

**Discussion by ZCH2005** [eclipse-langium/langium#1502] [Q&A] [answered]
I put the [Lox grammar](https://github.com/TypeFox/langium-lox/blob/main/src/language-server/lox.langium) in Langium playground and found that the member call to access class member failed. How to write the correct rule to access class member?

![image](https://github.com/eclipse-langium/langium/assets/2245424/d470b84b-4f36-4858-837e-1396b2c6a442)

**Comment by cdietrich** [comment, marked as answer]
in the playground you cannot implement scoping. so i would consider this expected

**Comment by msujew** [comment]
@ZCH2005 we have a [guide on our website](https://langium.org/docs/recipes/scoping/class-member/) that explains how the lox scoping works with links to its implementation.

Note that these scoping rules are not part of the grammar but defined in code as they go beyond the logic that can be expressed using EBNF.

**Reply by ZCH2005** [reply]
Hi @cdietrich and @msujew,

Thanks for answer. Now I understand that it's the limitation of the grammar.
